src/image.rs

```diff
@@ -183,11 +179,12 @@ impl ImageClient {
 
         // If one of self.config.auth and self.config.security_validate is enabled,
         // there will establish a secure channel between image-rs and Attestation-Agent
+        #[cfg(feature = "getresource")]
```

Comment:
Seems it may help readability by grouping code between line 182-263 as a helper function?

Reply:
As `secure_channel` of line 182 will also be used in line 289, and between them there should be other irrelevant code. If a function, there should be two different functions. One of them return a tuple `(SecureChannel, Auth)`, and the other return `Auth` controlled by feature, which will be more confusing. I also have not thought of a better way to refactor : (